bin/faucet/src/server/get_tokens.rs

```diff
@@ -181,10 +188,45 @@ impl RawMintRequest {
     }
 }
 
+struct RequestCounterGuard {
+    active_count: Arc<AtomicUsize>,
+    queued_count: Arc<AtomicUsize>,
+}
+
+impl RequestCounterGuard {
+    fn new(active_count: Arc<AtomicUsize>, queued_count: Arc<AtomicUsize>) -> Self {
+        active_count.fetch_add(1, Ordering::Relaxed);
+        RequestCounterGuard { active_count, queued_count }
+    }
+
+    fn mark_queued(&self) {
+        self.queued_count.fetch_add(1, Ordering::Relaxed);
+    }
+
+    fn mark_processing(&self) {
+        self.queued_count.fetch_sub(1, Ordering::Relaxed);
+    }
```

Comment:
Just wondering if it would make more sense to have two guards rather than these associated fns.

A guard for queued and a guard for processing. Might be a reason why that wouldn't work. But feels like it could be less error prone / risk of edge cases that we aren't unit testing for. Unsure.

Reply:
Perhaps `mark_enqueued` and `mark_dequeued` might be a bit more obvious for the reader?

Reply:
Wondering we don't mark dequeued when error occurs too.

Reply:
I simplified the requests tracking, having only active requests since I wasn't using the actual value of queued requests anyways.